frontend/src/app/data-entry/gift-form/gift-identification-form/gift-identification-form.component.ts

```diff
@@ -0,0 +1,105 @@
+import { Component, DestroyRef, OnInit } from "@angular/core";
+import { takeUntilDestroyed } from "@angular/core/rxjs-interop";
+import { FormGroup, FormControl, Validators } from "@angular/forms";
+import { ActivatedRoute } from "@angular/router";
+import { ToastService } from "@services/toast.service";
+import {
+    DataEntryGiftIdentificationGQL,
+    DataEntryUpdateGiftGQL,
+} from "generated/graphql";
+import {
+    map,
+    switchMap,
+    shareReplay,
+    filter,
+    debounceTime,
+    withLatestFrom,
+} from "rxjs";
+
+@Component({
+    selector: "lc-gift-identification-form",
+    templateUrl: "./gift-identification-form.component.html",
+    styleUrls: ["./gift-identification-form.component.scss"],
+})
+export class GiftIdentificationFormComponent implements OnInit {
+    public id$ = this.route.params.pipe(map((params) => params["id"]));
+
+    public gift$ = this.id$.pipe(
+        switchMap((id) => this.giftQuery.watch({ id }).valueChanges),
+        map((result) => result.data.giftDescription),
+        shareReplay(1)
+    );
+
+    public form = new FormGroup({
+        name: new FormControl("", {
+            validators: [Validators.required],
+            nonNullable: true,
+        }),
+        description: new FormControl("", {
+            nonNullable: true,
+        }),
+    });
+
+    constructor(
+        private destroyRef: DestroyRef,
+        private route: ActivatedRoute,
+        private toastService: ToastService,
+        private giftQuery: DataEntryGiftIdentificationGQL,
+        private giftMutation: DataEntryUpdateGiftGQL
+    ) {}
+
+    ngOnInit(): void {
+        this.gift$
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((gift) => {
+                if (!gift) {
+                    return;
+                }
+                this.form.patchValue(gift, {
+                    emitEvent: false,
+                    onlySelf: true,
+                });
+            });
+
+        this.gift$
+            .pipe(
+                switchMap(() =>
+                    this.form.valueChanges.pipe(
+                        map(() => this.form.getRawValue()),
```

Comment:
What does this do? I though this was already the payload of `form.valueChanges`?

Reply:
By default, you can disable any control in a FormGroup and their value won't show up in `form.value` (which is passed by `form.valueChanges`). As a result, `form.value`  is typed with `Partial<...>` to reflect the fact that all properties are optional. `.getRawValue()` takes the value of all controls, disabled or not, so the value has the type expected by the mutation.